secsgem/gem/hosthandler.py

```diff
@@ -139,7 +139,16 @@ def go_online(self):
         self.logger.info("Go online")
 
         # send remote command
-        return self.secs_decode(self.send_and_waitfor_response(self.stream_function(1, 17)())).get()
+        resp = self.secs_decode(self.send_and_waitfor_response(self.stream_function(1, 17)()))
```

Comment:
Why is the response of the message handled here? Shouldn't this be done in the calling function to allow flexibility with the call?

Reply:
I'd rather have it return None if it the reply not valid.

And I'd leave the interpretation of the value up to the calling function, as somebody might want to test the equipment for `secsgem.secs.data_items.ONLACK.NOT_ALLOWED`. An exception in this case interrupts the flow quite abruptly.
As every equipment might diverge from the standard, result interpretation should be left to the upper level user.

Reply:
I agree. I simplified the code accordingly.